docs/contentlayer.config.ts

```diff
@@ -88,6 +94,66 @@ export default makeSource({
     rehypePlugins: [
       [rehypeComponentDemo, { contentDirPath }],
       rehypeSlug,
+      // to inject the source code and other stuff inside `pre` element props
+      () => tree => {
+        visit(tree, node => {
+          if (node?.type === 'element' && node?.tagName === 'pre') {
+            const [codeEl] = node.children;
+            if (codeEl.tagName !== 'code') {
+              return;
+            }
+
+            if (codeEl.data?.meta) {
+              // Extract event from meta and pass it down the tree.
+              const regex = /event="([^"]*)"/;
+              const match = codeEl.data?.meta.match(regex);
+              if (match) {
+                node.__event__ = match ? match[1] : null;
+                codeEl.data.meta = codeEl.data.meta.replace(regex, '');
+              }
+            }
+            node.__rawString__ = codeEl.children?.[0].value;
+          }
+        });
+      },
+      [rehypeAutolinkHeadings, { behavior: 'wrap' }],
+      [
+        rehypePrettyCode,
+        {
+          getHighlighter: async () => {
+            const theme = await loadTheme(
+              // loading theme file
+              path.join(process.cwd(), './light.json')
```

Comment:
this could fail if `cwd` is not `/docs`, this is from the documentation

```js
const options = {
  theme: JSON.parse(
    readFileSync(new URL('./themes/theme.json', import.meta.url))
  ),
};
```